src/nodejs/hooks/seeker_agent_test.go

```diff
@@ -0,0 +1,479 @@
+				err = seeker.AfterCompile(stager)
+				Expect(err).To(BeNil())
+				contents, err := ioutil.ReadFile(entryPointPath)
+				Expect(err).To(BeNil())
+				Expect(string(contents)).To(Equal(
+					"require('./seeker/node_modules/@synopsys-sig/seeker-inline');\n" +
+						"some mock javascript code\n"))
+			})
+			It("Fails to prepend the require to the server.js file - when the file does not exist", func() {
+				entryPointPath := filepath.Join(buildDir, "server.js")
+				Expect(entryPointPath).ToNot(BeAnExistingFile())
+				err = seeker.AfterCompile(stager)
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).To(ContainSubstring("no such file or directory"))
+			})
+
+		})
+	})
+})
+
+func assertSensorDownload() {
+	Expect(httpmock.GetCallCountInfo()["GET "+getVersionURL()]).To(Equal(1))
+	Expect(httpmock.GetCallCountInfo()["GET "+getSensorURL()]).To(Equal(1))
+	Expect(httpmock.GetCallCountInfo()).ShouldNot(ContainElement("GET " + getAgentURL()))
+	Expect(httpmock.GetTotalCallCount()).To(Equal(2))
+}
+func assertAgentDownload() {
+	Expect(httpmock.GetCallCountInfo()["GET "+getVersionURL()]).To(Equal(1))
+	Expect(httpmock.GetCallCountInfo()["GET "+getAgentURL()]).To(Equal(1))
+	Expect(httpmock.GetCallCountInfo()).ShouldNot(ContainElement("GET " + getSensorURL()))
+	Expect(httpmock.GetTotalCallCount()).To(Equal(2))
+}
+
+func getAgentZip() ([]byte, error) {
+	return getFixtureContent("NODEJS_agent.zip")
+}
+func getSensorZip() ([]byte, error) {
+	return getFixtureContent("NODEJS_SensorWithAgent.zip")
+}
+func getFixtureContent(fileName string) ([]byte, error) {
+	path, err := filepath.Abs("../../../fixtures/seeker/" + fileName)
+	if err != nil {
+		return nil, err
+	}
+	fileContent, err := ioutil.ReadFile(path)
+	return fileContent, err
+}
+
+func mockSeekerVersionThatSupportsOnlySensorDownload() {
+	httpmock.RegisterResponder("GET", getVersionURL(),
+		httpmock.NewStringResponder(200, `{"publicName":"Seeker Enterprise Server","version":"2018.05-SP1-SNAPSHOT","buildNumber":"20180629131550","scmBranch":"origin/release/v2018.06","scmRevision":"815ba309"}`))
+}
+
+func mockSeekerVersionThatSupportsAgentDownload() {
+	httpmock.RegisterResponder("GET", getVersionURL(),
+		httpmock.NewStringResponder(200, `{"publicName":"Seeker Enterprise Server","version":"2018.06-SNAPSHOT","buildNumber":"20180629131550","scmBranch":"origin/release/v2018.06","scmRevision":"815ba309"}`))
+}
+func mockSensorDownload() {
+	zipContent, _ := getSensorZip()
+	httpmock.RegisterResponder("GET", getSensorURL(),
+		httpmock.NewBytesResponder(200, zipContent))
+}
+
+func mockAgentDownload() {
+	zipContent, _ := getAgentZip()
+	httpmock.RegisterResponder("GET", getAgentURL(),
+		httpmock.NewBytesResponder(200, zipContent))
+}
+func getSensorURL() string {
+	return "http://10.120.9.117:8082/rest/ui/installers/binaries/LINUX"
+}
+func getAgentURL() string {
+	return "http://10.120.9.117:8082/rest/ui/installers/agents/binaries/NODEJS"
+}
```

Comment:
Hey! When we ran the unit tests, this isn't the URL we see in the http request. We are seeing: `http://10.120.9.117:8082/rest/api/latest/installers/agents/binaries/NODEJS`

Some unit tests are failing on our end. We're seeing the tests on line 263, 322, 335, 394 fail. Did you mean to use the above url in these tests?

Reply:
I've refactor the tests.
The reason for different URLs for downloading the agent is that in older Seeker versions, the agents download URL was different. In order to be backward compatible we support both URLs depending on the Seeker version being used.